props/saber_BC_buttons.h

```diff
@@ -0,0 +1,1046 @@
+        last_push_ = millis();
+      }
+      return true;
+  #endif  // BC_FORCE_PUSH
+
+// Start or Stop Track
+  #if NUM_BUTTONS == 1
+    case EVENTID(BUTTON_POWER, EVENT_FOURTH_SAVED_CLICK_SHORT, MODE_OFF):
+    case EVENTID(BUTTON_POWER, EVENT_FOURTH_SAVED_CLICK_SHORT, MODE_ON):
+  #else
+    // 2 or 3 button
+    case EVENTID(BUTTON_POWER, EVENT_SECOND_SAVED_CLICK_SHORT, MODE_ON | BUTTON_AUX):
+    case EVENTID(BUTTON_POWER, EVENT_SECOND_SAVED_CLICK_SHORT, MODE_OFF | BUTTON_AUX):
+  #endif
+      if (!mode_volume_) {
+        StartOrStopTrack();
+      }
+      return true;
+
+// Next Preset AND Volume Up
+  #if NUM_BUTTONS == 1
+    case EVENTID(BUTTON_POWER, EVENT_FIRST_CLICK_LONG, MODE_ON):
+  #else
+  // 2 button
+    case EVENTID(BUTTON_POWER, EVENT_FIRST_CLICK_LONG, MODE_ON | BUTTON_AUX):
+  #endif
+    if (fusor.angle1() >  M_PI / 3) {
+      //Don't change preset if in colorchange mode
+      if (SaberBase::GetColorChangeMode() != SaberBase::COLOR_CHANGE_MODE_NONE) return false;
+        next_preset();
+      }
+      return true;
+    case EVENTID(BUTTON_POWER, EVENT_FIRST_CLICK_LONG, MODE_OFF):
+    if (!mode_volume_) {
+      next_preset();
+    } else {
+      VolumeUp();
+    }
+    return true;
+
+// Previous Preset AND Volume Down
+  #if NUM_BUTTONS == 1
+    case EVENTID(BUTTON_POWER, EVENT_SECOND_CLICK_LONG, MODE_ON):
+  #else
+  // 2 button
+    case EVENTID(BUTTON_POWER, EVENT_SECOND_CLICK_LONG, MODE_ON | BUTTON_AUX):
+  #endif
+    if (fusor.angle1() >  M_PI / 3) {
+      //Don't change preset if in colorchange mode
+      if (SaberBase::GetColorChangeMode() != SaberBase::COLOR_CHANGE_MODE_NONE) return false;
+        previous_preset();
+      }
+      return true;
+    case EVENTID(BUTTON_POWER, EVENT_SECOND_CLICK_LONG, MODE_OFF):
+      if (!mode_volume_) {
+        previous_preset();
+      } else {
+        VolumeDown();
+      }
+      return true;
+
+// Enter / Exit Volume MENU
+  #if NUM_BUTTONS == 1
+    case EVENTID(BUTTON_NONE, EVENT_CLASH, MODE_OFF | BUTTON_POWER):
+  #else
+    // 2 button
+    case EVENTID(BUTTON_AUX, EVENT_FIRST_CLICK_LONG, MODE_OFF):
+  #endif
+      if (!mode_volume_) {
+        mode_volume_ = true;
+        if (SFX_vmbegin) {
+          hybrid_font.PlayCommon(&SFX_vmbegin);
+        } else {
+          beeper.Beep(0.5, 3000);
+        }
+        STDOUT.println("Enter Volume Menu");
+      } else {
+        mode_volume_ = false;
+        if (SFX_vmend) {
+          hybrid_font.PlayCommon(&SFX_vmend);
+        } else {
+          beeper.Beep(0.5, 3000);
+        }
+        STDOUT.println("Exit Volume Menu");
+      }
+      return true;
+      
+// Spoken Battery Level
+  #if NUM_BUTTONS == 1
+    case EVENTID(BUTTON_POWER, EVENT_THIRD_SAVED_CLICK_SHORT, MODE_OFF):
+  #else
+    // 2 and 3 button
+    case EVENTID(BUTTON_AUX, EVENT_FIRST_HELD_LONG, MODE_OFF):
+  #endif 
+      if (SFX_battlevel) {        
+        hybrid_font.PlayCommon(&SFX_battlevel);
+        talkie.Say(spPAUSE1);
```

Comment:
This is bad.
I recommend writing this with a state machine instead.

Reply:
Bailed on the spoken stuff in light of pending OS6 edit mode.
Straight talkie now.